src/Modules/ExteriorPowers/FreeModules.jl

```diff
@@ -0,0 +1,361 @@
+    phi = wedge_multiplication_map(exterior_power(F, 0, cached=cached)[1], F, v)
+    return kernel(phi)[1]
+  end
+
+  if iszero(i) # Homology at the last entry of the complex.
+    phi = wedge_multiplication_map(exterior_power(F, n-1)[1], exterior_power(F, n, cached=cached)[1], v)
+    return cokernel(phi)[1]
+  end
+
+  ext_powers = [exterior_power(F, p, cached=cached)[1] for p in i-1:i+1]
+  boundary_maps = [wedge_multiplication_map(ext_powers[p], ext_powers[p+1], v) for p in 1:2]
+  K = chain_complex(boundary_maps)
+  return homology(K, 1)
+end
+
+function koszul_homology(v::FreeModElem, M::ModuleFP, i::Int; cached::Bool=true)
+  F = parent(v)
+  n = rank(F)
+
+  # Catch the edge cases
+  if i == n # This captures the homological degree zero due to the convention of the chain_complex constructor
+    phi = wedge_multiplication_map(exterior_power(F, 0, cached=cached)[1], F, v)
+    K = chain_complex([phi])
+    KM = tensor_product(K, M)
+    return kernel(map(KM, 1))[1]
+  end
+
+  if iszero(i) # Homology at the last entry of the complex.
+    phi = wedge_multiplication_map(exterior_power(F, n-1)[1], exterior_power(F, n, cached=cached)[1], v)
+    K = chain_complex([phi])
+    KM = tensor_product(K, M)
+    return cokernel(map(K, 1)) # TODO: cokernel does not seem to return a map by default. Why?
+  end
+
+  ext_powers = [exterior_power(F, p, cached=cached)[1] for p in i-1:i+1]
+  boundary_maps = [wedge_multiplication_map(ext_powers[p], ext_powers[p+1], v) for p in 1:2]
+  K = chain_complex(boundary_maps)
+  return homology(K, 1)
+end
+
+function koszul_dual(F::FreeMod; cached::Bool=true)
+  success, M, p = is_exterior_power(F)
+  !success && error("module must be an exterior power of some other module")
+  return exterior_power(M, rank(M) - p, cached=cached)[1]
+end
+
+function koszul_duals(v::Vector{T}; cached::Bool=true) where {T<:FreeModElem}
+  isempty(v) && error("list of elements must not be empty")
+  all(u->parent(u) === parent(first(v)), v[2:end]) || error("parent mismatch")
+
+  F = parent(first(v))
+  success, M, p = is_exterior_power(F)
+  n = rank(M)
+  success || error("element must be an exterior product")
+  k = [findfirst(x->x==u, gens(F)) for u in v]
+  any(x->x===nothing, k) && error("elements must be generators of the module")
+  ind = [ordered_multi_index(r, p, n) for r in k]
+  comp = [ordered_multi_index([i for i in 1:n if !(i in indices(I))], n) for I in ind]
+  lin_ind = linear_index.(comp)
+  F_dual = koszul_dual(F, cached=cached)
+  results = [F_dual[j] for j in lin_ind]
+  for r in 1:length(v)
+    sign, _ = _wedge(ind[r], comp[r])
+    isone(sign) || (results[r] = -results[r])
+  end
+  return results 
+end
+
+function koszul_dual(v::FreeModElem; cached::Bool=true)
+  return first(koszul_duals([v], cached=cached))
+end
+
+function induced_map_on_exterior_power(phi::FreeModuleHom{<:FreeMod, <:FreeMod, Nothing}, p::Int;
```

Comment:
A similar function already exists for LieAlgebraModule.jl under the name `hom_exterior_power`. It would be great to have a unified interface for both types of modules.
This is not a request for you to change your proposal, I just wanted to mention it.
@fieker suggested to name this method simply `hom` and supply both exterior powers as arguments, as well as the home on the base modules